sdk/inkrecognizer/src/InkRecognizerClient.ts

```diff
@@ -0,0 +1,154 @@
+import { InkRecognitionResult } from './InkRecognitionResult';
+import { parseInkRecognitionResult } from './InkRecognitionUnitBuilder';
+
+/**
+ * Default options for the request sent to the InkRecognizer service.
+ * @param version - InkRecognizer Service API version to use.
+ * @param applicationType - The domain of the application (Writing or Drawing. The default is "Mixed").
+ * @param language - IETF BCP 47 language code (for ex. en-US, en-GB, hi-IN etc.) for the strokes.
+ * @param unit - The physical unit for the points in the stroke. The default is "Millimeter".
+ * @param unitMultiple - A multiplier applied to the unit value to indicate the true unit being used.
+ */
+export interface InkRecognizerClientOptions extends ServiceClientOptions {
+  version: ServiceVersion;
+  applicationType: ApplicationKind;
+  language: string;
+  unit: InkPointUnit;
+  unitMultiple: number;
+}
+
+/**
+ * Per-request options to override the default options of the RecognizeInk request.
+ * @param applicationType - The domain of the application (Writing or Drawing. The default is "Mixed").
+ * @param language - IETF BCP 47 language code (for ex. en-US, en-GB, hi-IN etc.) for the strokes.
+ * @param unit - The physical unit for the points in the stroke. The default is "Millimeter".
+ * @param unitMultiple - A multiplier applied to the unit value to indicate the true unit being used.
+ */
+export interface RecognizeInkOptions {
+  applicationType: ApplicationKind;
+  language: string;
+  unit: InkPointUnit;
+  unitMultiple: number;
+}
+
+/**
+ * The InkRecognizerClient communicates with the service using default
+ * configuration settings or settings provided by the caller.
+ */
+export class InkRecognizerClient extends ServiceClient {
+  // private readonly baseUri: string;
+  private readonly url: string;
+  private readonly credentials: ApiKeyCredentials;
+
+  private readonly applicationType: ApplicationKind;
+  private readonly language: string;
+  private readonly version: ServiceVersion;
+  private readonly unit: InkPointUnit;
+  private readonly unitMultiple: number;
+
+  /**
+  * Initializes a new instance of the InkRecognizerClient class.
+  * @param baseUri - The url of the InkRecognizer Service
+  * @param credentials - Subscription credentials which uniquely identify client subscription.
+  * @param [options] - The parameter options
+  */
+  constructor(baseUri: string,
+    credentials: ApiKeyCredentials,
+    options?: InkRecognizerClientOptions) {
+
+    if (baseUri == undefined) {
+      throw new Error("'url' is invalid.");
+    }
+    if (credentials === undefined) {
+      throw new Error("'credentials' can not be empty.");
+    }
+    if (!options) {
+      options = {} as InkRecognizerClientOptions;
+      options.applicationType = ApplicationKind.Mixed;
+      options.language = 'en-US';
+      options.version = ServiceVersion.Preview_1_0_0;
+      options.unit = InkPointUnit.Millimeter;
+      options.unitMultiple = 1.0;
```

Comment:
Shouldn't we be setting the user agent here as well?
Below is a code snippet from another library
```javascript
if(!options.userAgent) {
      const defaultUserAgent = coreHttp.getDefaultUserAgentValue();
      options.userAgent = `${packageName}/${packageVersion} ${defaultUserAgent}`;
    }
```

Reply:
Wonder what is the need of useragent here? Our service is browser independent, do we still need to set this?

Reply:
It is mainly used to determine which version of the library is being used to make the service calls. Helps with determining usage across time.

Reply:
Thanks, this will be very useful to know how many users are using SDK, will add it.